Maintenance

grturner opened the pull request:
Corrected some vulnerabilities and code smells.

grturner committed:
Maintenance: DTO's
Implemented Data Transfer Objects to correct some vulnerabilities found by SonarCloud.

grturner committed:
Code Smells
Code was running the get method on Optionals without first checking if present. Refactored the Services to simply pass the Optional back to the controller where the controller can perform logic on the optional and return proper http error codes.

IkenoXamos commented on src/main/java/com/revature/controllers/AdminController.java:
```diff
@@ -37,76 +28,71 @@
-		return as.getAdminById(id);
+	public ResponseEntity<Admin> getAdminById(@PathVariable("id") int id) {
+		Optional<Admin> admin = adminService.getAdminById(id);
+		return admin.map(value -> ResponseEntity.ok().body(value)).orElseGet(() -> ResponseEntity.notFound().build());
```
I think it would be more readable to move the .orElseGet to the next line.

grturner replied:
Resolved in 325399c

IkenoXamos commented on src/main/java/com/revature/controllers/BatchController.java:
```diff
@@ -38,69 +28,65 @@
+	@GetMapping("/{id}")
+	public ResponseEntity<Batch> getBatchByNumber(@PathVariable("id") int id) {
+		Optional<Batch> batch = batchService.getBatchByNumber(id);
+		return batch.map(value -> ResponseEntity.ok().body(value)).orElseGet(() -> ResponseEntity.notFound().build());
```
Collapse into 2 lines for all similar Controller methods.

grturner replied:
Resolved in 325399c

IkenoXamos commented on src/main/java/com/revature/controllers/LoginController.java:
```diff
@@ -43,49 +18,46 @@
-		 info.computeIfAbsent("googleMapAPIKey", key -> new HashSet<>()).add(newkey);
-		 return info;
-	}
+        Map<String, Set<String>> info = new HashMap<>();
```
Tabs preferable to spaces for alignment.

grturner replied:
Resolved in 325399c

IkenoXamos commented on src/main/java/com/revature/models/Car.java:
```diff
@@ -39,25 +39,24 @@
-		this.year = year;
-		this.user = user;
-	}
+    private String model;
```
Tabs preferable to spaces.

grturner replied:
Resolved in 325399c

IkenoXamos commented on src/main/java/com/revature/models/CarDTO.java:
```diff
@@ -0,0 +1,26 @@
+@Data
+@NoArgsConstructor
+public class CarDTO {
+    private int carId;
```
Should check all DTOs for spaces vs tabs. Either tabs or spaces would work, but since this repository is using tabs in the large majority of locations, new code should stay consistent.

grturner replied:
Resolved in 325399c

IkenoXamos commented on src/main/java/com/revature/services/AdminService.java:
```diff
@@ -1,14 +1,19 @@
-	public Admin updateAdmin(Admin admin);
-	public String deleteAdminById(int id);
+
+    List<Admin> getAdmins();
```
Tabs

IkenoXamos replied:
Check all services for tabs/spaces

grturner replied:
Resolved in 325399c

grturner committed:
Requested Changes
Implemented requested changes and also included updated tests that work with our Data Transfer Objects.

IkenoXamos reviewed (approved):
LGTM
